### PR TITLE
feat(cli): allow lib to be used using cli

### LINKED ISSUE #3: Drop the gulp dependency
I think this should also be delivered as a cli-tool and we should be able to run it directly. dont you think ?

```
check-deps --fail-level=minor
```

### PR DESCRIPTION
**Pull request by iamdey**
should work with global installation
- fix CS

closes #3

### CODE REVIEW

**Comment by iamdey** [comment]
Just checked, global install works great

**Comment by jubianchi** [comment]
I :heart: it!

**Comment by jubianchi** [comment]
I would like to go further because what you did is awesome but we still require gulp for some things and they may not be required when using the CLI mode.

What do you think of splitting this in two NPM packages :
- check-deps
- gulp-check-deps

We can do that and keep a single code repository by creating a subfolder to hold gulp specific code.

Here is how I see this:
- [ ] rename the root package: `gul-check-deps` :arrow_right: `check-deps`
- [ ] bump the root package to `2.0.0`
- [ ] create a `gulp` subfolder holding the code required to call the library and emit gulp events
- [ ] add a `package.json` inside the `gulp`  subfolder with gulp specific dependencies and a dependencies on `check-deps`
- [ ] name it `gulp-check-deps`
- [ ] bump the `gulp-check-deps` package to `1.(y+1).0`
- [ ] rework the root readme and add a new one for the sub-package `gulp-check-deps`
- [ ] publish both to NPM

If this todo-list seems good to you, I'll create the associated issue for later resolution. I'll merge your actual PR and release a `1.5.0` because the feature as-is is still relevant.

**Comment by iamdey** [comment]
Yeah I'm totally agree with your TODO list.

**Comment by Mandragora2** [comment]
:heart: it, i agree with @jubianchi for split into split into 2 npm packages
really great work @iamdey ;)

File: bin/check-deps.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    name: 'prerelease',
+    short: 'r',
+    type: 'boolean',
+    description: 'Fail for pre-release'
+  },
+  {
+    name: 'level',
+    short: 'l',
+    type: 'int',
+    description: 'Fail Level in: [' + Object.keys(checkDeps.levels).map(function(k) {
```
Comment (Mandragora2) [inline bin/check-deps.js]: Replace `k` by `key` for not using single letter var ? ^^

Comment (iamdey) [inline bin/check-deps.js]: you right

**Comment by iamdey** [comment]
please merge to make #7 more readable

**Comment by iamdey** [comment]
I forgot some `util.log(row);` I close this and go to #7

**Pull request closed without merging**